src/Development/IDE/Core/Compile.hs

```diff
@@ -447,8 +449,46 @@ getModSummaryFromBuffer fp contents dflags parsed = do
     where
       sourceType = if "-boot" `isSuffixOf` takeExtension fp then HsBootFile else HsSrcFile
 
-
--- | Given a buffer, flags, file path and module summary, produce a
+-- | Given a buffer, env and filepath, produce a module summary by parsing only the imports.
+--   Runs preprocessors as needed.
+getModSummaryFromImports :: FilePath -> Maybe SB.StringBuffer -> HscEnv -> ExceptT [FileDiagnostic] IO ModSummary
+getModSummaryFromImports fp contents hsc_env = do
+    (contents, dflags) <- ExceptT $ evalGhcEnv hsc_env $ runExceptT $ preprocessor fp contents
+    (srcImports, textualImports, L _ moduleName) <-
+        ExceptT $ first (diagFromErrMsgs "parser" dflags) <$> GHC.getHeaderImports dflags contents fp fp
+    liftIO $ evaluate $ rnf srcImports
+    liftIO $ evaluate $ rnf textualImports
+    modLoc <- liftIO $ mkHomeModLocation dflags moduleName fp
+    required_by_imports <- liftIO $ implicitRequirements hsc_env textualImports
```

Comment:
Looks like this is copied from something to do with backpack, do we support backpack projects?

`implicitRequirements` calls `findImportedModule` which is probably desirable to avoid here.

Reply:
ack